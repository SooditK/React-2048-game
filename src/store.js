import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import undoable from 'redux-undo';
import rootReducer from './reducers';
import rootSaga from './sagas';

const initHistory = JSON.parse(localStorage.getItem('state') || 'null');

const sagaMiddleware = createSagaMiddleware();
/* eslint-disable no-underscore-dangle */
const args = [
  undoable(rootReducer, {
    limit: 11, // set a limit for the history
    ignoreInitialState: true
  })
];
// Load localStorage data if available
if (initHistory) {
  args.push(initHistory);
}

if (window.__REDUX_DEVTOOLS_EXTENSION__) {
  args.push(
    compose(
      applyMiddleware(sagaMiddleware),
      // Redux devtools necessary code
      window.__REDUX_DEVTOOLS_EXTENSION__()
    )
  );
} else {
  args.push(applyMiddleware(sagaMiddleware));
}

const store = createStore(...args);
sagaMiddleware.run(rootSaga);

// Call this function while redux state changed,
// this callback save redux state to localStorage
store.subscribe(() => {
  const state = store.getState();
  localStorage.setItem('state', JSON.stringify(state));
});

export default store;
