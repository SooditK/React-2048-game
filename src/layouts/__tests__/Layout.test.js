import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
import { Provider } from 'react-redux';
import store from '../../store';
import Layout from '..';

// Make random always return 0.5, so snapshot always same.
const mockMath = Object.create(global.Math);
mockMath.random = () => 0.5;
global.Math = mockMath;

describe('<Layout />', () => {
  it('component render', () => {
    const layout = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <Layout />
          </MemoryRouter>
        </Provider>
      )
      .toJSON();
    expect(layout).toMatchSnapshot();
  });
});
