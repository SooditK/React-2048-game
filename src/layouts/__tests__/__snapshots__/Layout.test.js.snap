// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> component render 1`] = `
<div>
  <header
    className="header"
  >
    <nav>
      <ul
        className="row"
      >
        <li
          className="title"
        >
          <a
            aria-current="true"
            className="active"
            href="/"
            onClick={[Function]}
            style={Object {}}
          >
            Home
          </a>
        </li>
        <li
          className="link"
        >
          <a
            aria-current={false}
            className={undefined}
            href="/ranking"
            onClick={[Function]}
            style={undefined}
          >
            Ranking
          </a>
        </li>
        <li
          className="link"
        >
          <a
            aria-current={false}
            className={undefined}
            href="/comments"
            onClick={[Function]}
            style={undefined}
          >
            Comments
          </a>
        </li>
        <li
          className="spacer"
        />
        <li>
          <a
            href="https://github.com/devrsi0n/React-2048-game"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="github repo"
              className="icon"
              src="github-white.png"
            />
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <main>
    <div>
      <div
        className="app"
      >
        <div
          className="box"
        >
          <div
            className="board"
          >
            <h1
              className="title"
            >
              2048
            </h1>
            <table
              className="board"
            >
              <tbody>
                <tr>
                  <td>
                    <div
                      className="cell"
                    >
                      <div
                        className="number"
                      />
                    </div>
                  </td>
                  <td>
                    <div
                      className="cell"
                    >
                      <div
                        className="number"
                      />
                    </div>
                  </td>
                  <td>
                    <div
                      className="cell"
                    >
                      <div
                        className="number"
                      />
                    </div>
                  </td>
                  <td>
                    <div
                      className="cell"
                    >
                      <div
                        className="number"
                      />
                    </div>
                  </td>
                </tr>
                <tr>
                  <td>
                    <div
                      className="cell"
                    >
                      <div
                        className="number"
                      />
                    </div>
                  </td>
                  <td>
                    <div
                      className="cell"
                    >
                      <div
                        className="number"
                      />
                    </div>
                  </td>
                  <td>
                    <div
                      className="cell"
                    >
                      <div
                        className="number"
                      />
                    </div>
                  </td>
                  <td>
                    <div
                      className="cell color-4"
                    >
                      <div
                        className="number"
                      >
                        4
                      </div>
                    </div>
                  </td>
                </tr>
                <tr>
                  <td>
                    <div
                      className="cell color-4"
                    >
                      <div
                        className="number"
                      >
                        4
                      </div>
                    </div>
                  </td>
                  <td>
                    <div
                      className="cell"
                    >
                      <div
                        className="number"
                      />
                    </div>
                  </td>
                  <td>
                    <div
                      className="cell"
                    >
                      <div
                        className="number"
                      />
                    </div>
                  </td>
                  <td>
                    <div
                      className="cell"
                    >
                      <div
                        className="number"
                      />
                    </div>
                  </td>
                </tr>
                <tr>
                  <td>
                    <div
                      className="cell"
                    >
                      <div
                        className="number"
                      />
                    </div>
                  </td>
                  <td>
                    <div
                      className="cell"
                    >
                      <div
                        className="number"
                      />
                    </div>
                  </td>
                  <td>
                    <div
                      className="cell"
                    >
                      <div
                        className="number"
                      />
                    </div>
                  </td>
                  <td>
                    <div
                      className="cell"
                    >
                      <div
                        className="number"
                      />
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
          <div
            className="panel"
          >
            <div>
              <div
                className="index"
              >
                <p
                  className="score"
                >
                  SCORE
                  <em
                    className="text"
                  >
                    0
                  </em>
                </p>
                <p
                  className="score best"
                >
                  BEST
                  <em
                    className="text"
                  >
                    0
                  </em>
                </p>
              </div>
            </div>
            <div
              className="panel"
            >
              <div
                className="rowBtn"
              >
                <div
                  className="btn"
                >
                  <button
                    className="btn-flat btn-md btn-default"
                    onClick={[Function]}
                  >
                    <img
                      alt="speaker"
                      src="speaker-on.svg"
                    />
                  </button>
                </div>
                <div
                  className="btn "
                >
                  <button
                    className="btn-flat btn-md btn-default"
                    onClick={[Function]}
                  >
                    <img
                      alt="undo"
                      src="undo.svg"
                    />
                  </button>
                </div>
                <div
                  className="btn "
                >
                  <button
                    className="btn-flat btn-md btn-default"
                    onClick={[Function]}
                  >
                    <img
                      alt="reset"
                      src="reset.svg"
                    />
                  </button>
                </div>
              </div>
              <div
                className="btn "
              >
                <button
                  className="btn-flat btn-md btn-primary"
                  onClick={[Function]}
                >
                  <img
                    alt="arrow up"
                    className="up"
                    src="arrow.svg"
                  />
                </button>
              </div>
              <div
                className="arrows"
              >
                <div
                  className="btn "
                >
                  <button
                    className="btn-flat btn-md btn-primary"
                    onClick={[Function]}
                  >
                    <img
                      alt="arrow left"
                      className="left"
                      src="arrow.svg"
                    />
                  </button>
                </div>
                <div
                  className="btn "
                >
                  <button
                    className="btn-flat btn-md btn-primary"
                    onClick={[Function]}
                  >
                    <img
                      alt="arrow down"
                      className="down"
                      src="arrow.svg"
                    />
                  </button>
                </div>
                <div
                  className="btn "
                >
                  <button
                    className="btn-flat btn-md btn-primary"
                    onClick={[Function]}
                  >
                    <img
                      alt="arrow right"
                      className="right"
                      src="arrow.svg"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="tips"
        >
          <p
            className="title"
          >
            Tips
          </p>
          <p
            className="content"
          >
            Use keyboard arrow keys (or \`WASD\` if you like) control blocks;Click undo button to revert to last step status if you regrets。You score will upload to ranking list after you login at \`Comments\` panel
          </p>
        </div>
        <div>
          <div
            className="hidden"
          >
            <div
              className="overlay"
            />
            <div
              className="modal center"
            >
              <div>
                <div
                  className="gameover"
                >
                  <div
                    className="text"
                  >
                     
                    SCORE
                     
                  </div>
                  <div
                    className="banner"
                  >
                    <p>
                      0
                    </p>
                  </div>
                </div>
                <div
                  className="buttonWrapper"
                >
                  <button
                    className="button"
                    onClick={[Function]}
                  >
                    <img
                      alt="reset"
                      src="reset.svg"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
  <footer
    style={
      Object {
        "margin": "auto",
        "width": "90vw",
      }
    }
  >
    <div
      className="footer"
    >
      <div
        className="container"
      >
        <div>
          <a
            className="icon"
            href="https://github.com/devrsi0n/React-2048-game"
          >
            <img
              alt="github repo"
              className="github"
              src="github.png"
            />
          </a>
        </div>
        Build with
        <span
          className="heart"
        >
          ♥
        </span>
        by
        <a
          className="link"
          href="https://github.com/devrsi0n"
        >
          devrsi0n
        </a>
      </div>
    </div>
  </footer>
</div>
`;
